src/Operators/finitedifference.jl

```diff
@@ -1787,7 +1989,182 @@ Base.@propagate_inbounds function stencil_right_boundary(
+where h and l superscripts represent the high and lower order (monotone) 
+fluxes respectively. The effect of the TVD limiters is then to 
+adjust the flux 
+F_{j+1/2} = F^{l}_{j+1/2} + C_{j+1/2}(F^{h}_{j+1/2} - F^{l}_{j+1/2})
+where C_{j+1/2} is the multiplicative limiter which is a function of 
+the ratio of the slope of the solution across a cell interface.
+C=1 recovers the high order flux.
+C=0 recovers the low order flux. 
+
+Supported limiter types are 
+- RZeroLimiter (returns low order flux)
+- RHalfLimiter (flux multiplier == 1/2)
+- RMaxLimiter (returns high order flux)
+- MinModLimiter
+- KorenLimiter
+- SuperbeeLimiter
+- MonotonizedCentralLimiter
+"""
+abstract type AbstractTVDSlopeLimiter end
+struct RZeroLimiter <: AbstractTVDSlopeLimiter end
+struct RHalfLimiter <: AbstractTVDSlopeLimiter end
+struct RMaxLimiter <: AbstractTVDSlopeLimiter end
+struct MinModLimiter <: AbstractTVDSlopeLimiter end
+struct KorenLimiter <: AbstractTVDSlopeLimiter end
+struct SuperbeeLimiter <: AbstractTVDSlopeLimiter end
+struct MonotonizedCentralLimiter <: AbstractTVDSlopeLimiter end
+
+@inline function compute_limiter_coeff(r, ::RZeroLimiter)
+    return zero(eltype(r))
+end
+
+@inline function compute_limiter_coeff(r, ::RHalfLimiter)
+    return one(eltype(r)) * 1 / 2
+end
+
+@inline function compute_limiter_coeff(r, ::RMaxLimiter)
+    return one(eltype(r))
+end
+
+@inline function compute_limiter_coeff(r, ::MinModLimiter)
+    return max(zero(eltype(r)), min(one(eltype(r)), r))
+end
+
+@inline function compute_limiter_coeff(r, ::KorenLimiter)
+    return max(zero(eltype(r)), min(2r, min(1 / 3 + 2r / 3, 2)))
+end
+
+@inline function compute_limiter_coeff(r, ::SuperbeeLimiter)
+    return max(zero(eltype(r)), min(one(eltype(r)), r), min(2, r))
+end
+
+@inline function compute_limiter_coeff(r, ::MonotonizedCentralLimiter)
+    return max(zero(eltype(r)), min(2r, (1 + r) / 2, 2))
+end
+
+struct TVDSlopeLimitedFlux{BCS} <: AdvectionOperator
+    bcs::BCS
+end
+
+TVDSlopeLimitedFlux(; method, kwargs...) =
+    TVDSlopeLimitedFlux((; method, kwargs...))
+
+return_eltype(::TVDSlopeLimitedFlux, A, Φ, 𝓊) =
+    Geometry.Contravariant3Vector{eltype(eltype(A))}
+
+return_space(
+    ::TVDSlopeLimitedFlux,
+    A_space::AllFaceFiniteDifferenceSpace,
+    Φ_space::AllCenterFiniteDifferenceSpace,
+    𝓊_space::AllFaceFiniteDifferenceSpace,
+) = A_space
+
+function tvd_limited_flux(Aⱼ₋₁₂, Aⱼ₊₁₂, ϕⱼ₋₁, ϕⱼ, ϕⱼ₊₁, ϕⱼ₊₂, rⱼ₊₁₂, method)
+    stable_zero = zero(eltype(Aⱼ₊₁₂))
+    stable_one = one(eltype(Aⱼ₊₁₂))
+    Cⱼ₊₁₂ = compute_limiter_coeff(rⱼ₊₁₂, method)
+    @assert Cⱼ₊₁₂ <= eltype(Aⱼ₊₁₂)(2)
+    @assert Cⱼ₊₁₂ >= eltype(Aⱼ₊₁₂)(0)
+    return Cⱼ₊₁₂ * Aⱼ₊₁₂
+end
+
+stencil_interior_width(::TVDSlopeLimitedFlux, A_space, Φ_space, 𝓊_space) =
+    ((-1, 1), (-half - 1, half + 1), (-1, +1))
+
+Base.@propagate_inbounds function stencil_interior(
+    op::TVDSlopeLimitedFlux,
+    loc,
+    space,
+    idx,
+    hidx,
+    A_field,
+    Φ_field,
+    𝓊_field,
```

Comment:
We might want to modify this so that we can pass `dt` separately, as in the case of the VanLeer limiter. I think that would be somewhat clearer, but we can keep the arguments as they are now if it's more standard.

Reply:
Yes - currently this choice of args is also true in the `Zalesak` limiter - I can switch this in a future PR to assign `dt` as a separate argument.

Reply:
For the sake of getting this PR merged, let's follow this up later.

Reply:
Opened https://github.com/CliMA/ClimaCore.jl/issues/2107 to track.